src/compiler/checker.ts

```diff
@@ -11333,8 +11340,16 @@ namespace ts {
                 }
                 getTypeOfAccessors(getSymbolOfNode(node));
             }
+            if (node.parent.kind !== SyntaxKind.ObjectLiteralExpression) {
+                checkSourceElement(node.body);
+            }

```

Comment:
Does `checkFunctionLikeDeclaration` ever check accessors now? If not, then the line `if(!isAccessor(node.kind) && !node.asteriskToken)` can become `if(!node.asteriskToken)`.